corpus/convert.py

```diff
@@ -93,6 +106,59 @@ def run(self):
         c = corpus.Corpus()
         c.load(corpus_path)
 
-        with open(self.txt_path.get_path(), "wt") as out:
-            for segment in c.segments():
+        if self.segment_file:
+            with open(tk.uncached_path(self.segment_file), "rt") as f:
+                lines = [l for l in f.readlines() if len(l.strip()) > 0]
+                segments_whitelist = set(lines)
+        else:
+            segments_whitelist = None
+
+        if self.gzip:
+            out = gzip.open(self.out_txt.get_path(), "wt")
+        else:
+            out = open(self.out_txt.get_path(), "wt")
+
+        for segment in c.segments():
+            if not segments_whitelist or segment.fullname in segments_whitelist:
                 out.write(segment.orth + "\n")
+
+        out.close()
+
+
+class CorpusReplaceOrthFromTxtJob(Job):
+    """
+    Merge raw text back into a bliss corpus
+    """
+
+    def __init__(self, corpus, text_file):
```

Comment:
This Job should work in tandem with the CorpusToTxtJob, i.e. support a list of segments to be passed into it as it relies on the line-number to match transcriptions to segments.

Reply:
I just copied the job from an old setup, but yes I could add that functionality.

Reply:
:+1: